mleap-core/src/main/scala/ml/combust/mleap/core/types/DataType.scala

```diff
@@ -131,3 +131,14 @@ case class ListType(override val base: BasicType,
 
   override def shape: ListShape = ListShape(isNullable)
 }
+
+
+case class MapType(val key: BasicType,
+                   override val base: BasicType,
+                   override val isNullable: Boolean = true) extends DataType {
+
+  override def setNullable(isNullable: Boolean): DataType = copy(isNullable = isNullable)
+  override def simpleString: String = "map"
+  override def printString: String = s"$simpleString(key=$key,base=$base,nullable=$isNullable)"
+  override def shape: ListShape = ListShape(isNullable)
```

Comment:
I used LIstShape here because I think the semantics are the same (i.e., it is a collection that can have a variable length).  Open to discussion though

Reply:
We aren't serializing MapType anywhere.  The bundle just has to serialize the shape and node objects right?

Reply:
Had some offline discussions, and agreed we don't need a MapShape for now.